### PR TITLE
Remove the dependency on System.ValueTuple

### PR DESCRIPTION
**Pull request by SergeyTeplyakov**
Ok, this one is more interesting than the previous ones.

There are two reasons why I can't directly use the demystifier in my project:

* Dependency on System.ValueTuple
* Dependency on System.Reflection.Metadata

There are no issues with the dependency itself, but due to a small versioning hell that is happening right now with ValueTuple the dependency can cause lots of issues. (I know that at least one team that works on Visual Studio extension had to fork the code to remove the dependency on ValueTuple).

This pr removes the dependency and uses ... reflection (the dependency should be restored in the future, IMO).

I've tested the performance implication of such a change (the first commit of the PR has the benchmark that I was using):

TLDR; the performance difference is negligible (the following table actually shows that reflection-based version is faster; this is a clear indication that the difference is barely noticeable).

```
                                                    Method |  Job | Runtime |         Mean |        Error |       StdDev |       Median |   Scaled | ScaledSD |   Gen 0 | Allocated |
---------------------------------------------------------- |----- |-------- |-------------:|-------------:|-------------:|-------------:|---------:|---------:|--------:|----------:|
                                               .ToString() |  Clr |     Clr |   2,471.6 ns |     93.68 ns |    271.80 ns |   2,321.3 ns |     1.00 |     0.00 |  0.2289 |     720 B |
                                    Demystify().ToString() |  Clr |     Clr |  28,795.9 ns |    569.39 ns |    584.72 ns |  28,542.3 ns |    11.78 |     1.17 |  2.6855 |    8504 B |
                                '(left, right).ToString()' |  Clr |     Clr |  89,760.2 ns |  1,505.10 ns |  1,334.24 ns |  89,687.3 ns |    36.71 |     3.61 |  1.4648 |    4705 B |
 '(left, right).Demystify(useReflection: true).ToString()' |  Clr |     Clr | 912,079.5 ns | 16,122.25 ns | 14,291.95 ns | 907,792.3 ns |   372.98 |    36.75 | 16.6016 |   53760 B |
 '(left, right).Demystify(useReflection:false).ToString()' |  Clr |     Clr | 923,591.5 ns | 25,396.90 ns | 22,513.69 ns | 917,117.9 ns |   377.69 |    37.84 | 16.6016 |   53679 B |
```

### CODE REVIEW

**Comment by benaadams** [comment]
Thank you!

**Comment by benaadams** [comment]
In version 0.1.1 https://www.nuget.org/packages/Ben.Demystifier/0.1.1
